src/algebra/regular.lean

```diff
@@ -205,6 +205,28 @@ lemma is_regular_iff_subsingleton : is_regular (0 : R) ↔ subsingleton R :=
 ⟨λ h, h.left.subsingleton,
  λ h, ⟨is_left_regular_zero_iff_subsingleton.mpr h, is_right_regular_zero_iff_subsingleton.mpr h⟩⟩
 
+/--  A left-regular element of a `nontrivial` `mul_zero_class` is non-zero. -/
+lemma is_left_regular.ne_zero [nontrivial R] (la : is_left_regular a) : a ≠ 0 :=
+begin
+  rintro rfl,
+  rcases exists_pair_ne R with ⟨x, y, xy⟩,
+  refine xy (la _),
+  rw [zero_mul, zero_mul]
+end
+
+/--  A right-regular element of a `nontrivial` `mul_zero_class` is non-zero. -/
+lemma is_right_regular.ne_zero [nontrivial R] (ra : is_right_regular a) : a ≠ 0 :=
+begin
+  rintro rfl,
+  rcases exists_pair_ne R with ⟨x, y, xy⟩,
+  refine xy (ra (_ : x * 0 = y * 0)),
```

Comment:
Is the annotation needed here? You don't have it for the other lemma
```suggestion
  refine xy (ra _),
```

Reply:
Actually, it is needed: the rw does not see through defeq and the multiplication on the right gives out a weird expression, contrary to left multiplication.

Reply:
Had I proved it with simp, I would probably not have needed the type annotations.

Reply:
Ah right, it's the partial application in the injectivity differing between the two cases; makes sense. `delta` would work, but what you have is fine.